docs/architecture/architecture.md

```diff
@@ -6,48 +6,71 @@ This document describes the architecture of ibc-rs. If you're looking for a high
 
 ![](docs/architecture/assets/repo-context.png)
 
-[What is it core reason developers use Hermes for?]
+At its highest level, `ibc-rs` defines the InterBlockChain protocol via [specifications](ibc-specs) and exposes modules that implement the specified protocol logic. The IBC protocol can be split between on-chain and off-chain processes. The main off-chain component is Hermes, which is a packet relayer that facillitates communication between two chains. The main on-chain components deal with the concepts of clients, connections, and channels. 
 
-At its highest level, ibc-rs [does what?]
+## How to Read the Codebase
 
+`ibc-rs` (as well as many of Informal's other codebases) is structured a little bit differently from what you might see in a more "traditional" open source codebase. The main differentiating factor is the fact that the different pieces and components are specified in [standards](ibc-standards).
+
+Answer how to read the codebase, i.e. `icsxx_`
 
 ## Code Map 
 
-This section talks briefly about the various directories and modules in ibc-rs. 
+This section talks briefly about the various directories and modules in `ibc-rs`. 
 
 ### `modules`
 
-The main data structures and on-chain logic of the IBC protocol.
+This crate contains the main data structures and on-chain logic of the IBC protocol. These are the fundamental pieces that make up the IBC protocol. There is the conceptual notion of 'handlers', which are pieces of code that each handle a particular type of message. The most notable handlers are the [client](ibc-client), [connection](ibc-connection), and [channel](ibc-channel) handlers.  
+
+#### Clients
+
+Clients encapsulate all of the verification methods of another IBC-enabled chain in order to ensure that the other chain adheres to the IBC protocol and does not exhibit misbehaviour. Clients "track" the metadata of the other chain's blocks, and each chain has a client for every other chain that it communicates with. 
+
+#### Connections
+
+Connections associate a chain with another chain by connecting a client on the local chain with a client on the remote chain. This association is pair-wise unique and is established between two chains following a 4-step handshake process. 
+
+#### Channels
+
+Channels are an abstraction layer that facilitate communication between applications and the chains those applications depend upon. One important function that channels fulfill is guaranteeing that data packets sent between an application and its chain are well-ordered. 
 
 ### `relayer`
 
-Provides the logic for relaying datagrams between chains. 
+This crate provides the logic for relaying datagrams between chains. The process of relaying packets is kicked off by submitting transactions to read from or write to an IBC-enabled chain's state. More broadly, a relayer enables a chain to ascertain another chain's state by accessing its clients, connections, channels, or anything that is IBC-related.
```

Comment:
Maybe mention is an off-chain process, to link it back to the overview at the top of this doc?